File: PepcodingSept_19/Lec_002(factorialNCRAndRotate)/RotateNumber.cpp
#include<iostream>
#include<math.h>

using namespace std;
int digit(int n)
{
    int i=1;
    int a=n;
    while((a/10)!=0)
     {
       i++;
       a=a/10;
     }
     return i;
}
int rotate(int n,int r)
{
    int b,c;
    c=digit(n);
    b=r%c;
    b=(b<0)?(b+c):b;
    int multip=1;
    int divd=1;
    for(int i=1;i<=c;i++)
     {
       if(i<=r)
       multip*=10;
       else 
       divd*=10;
     }  
     
    // multip=pow(10,r)+1;
    // divd=pow(10,(c-r))+1;
    // cout<<multip<<" "<<divd<<" ";
    int fdigit=n%divd;
    int sdigit=n/divd;
     

    int num=fdigit*multip+sdigit;
    return num;
   
}
int main(int args,char** argv)
{
    int n,r;
    cout<<"Enter the number :";
    cin>>n;   
    cout<<"Enter r:";
    cin>>r;
    cout<<endl<<"Number after rotation:";
    cout<<rotate(n,r);
   
    return 0;
}

File: PepcodingSept_19/Lec_002(factorialNCRAndRotate)/RotateNumber.java
import java.util.Scanner;
public class RotateNumber {

  public static Scanner scn = new Scanner(System.in);
  public static void main(String[] args) {
    int n=0,r=0;
    System.out.print("Enter the number: ");
    n=scn.nextInt();
    System.out.print("Enter the number of digits to be rotated clockwise: ");
    r=scn.nextInt();
    System.out.print("The number after rotation is: " + rotate(n,r));
  }
  public static int rotate(int n,int r) {

    int digits=0,multipler=1,divisor=1,fdigit=0,sdigit=0;
    digits=digit(n);
    r=r%digits;
    r=(r<0)?(r+digits):r;
    for(int i=1;i<=digits;i++) {

       if(i<=r) {

        multipler*=10;
        }
       else {

        divisor*=10;
       }
     }
    fdigit=n%divisor;
    sdigit=n/divisor;

    return fdigit*multipler + sdigit;

  }
  public static int digit(int n) {
    
    int ans=0;
    while(n!=0) {
      ans++;
      n/=10;
    }
    return ans;
  }
}